piptools/scripts/compile.py

```diff
@@ -302,6 +303,20 @@ def _determine_linesep(
     help="Specify a package to consider unsafe; may be used more than once. "
     f"Replaces default unsafe packages: {', '.join(sorted(UNSAFE_PACKAGES))}",
 )
+@click.option(
+    "--config",
+    type=click.Path(
+        exists=True,
+        file_okay=True,
+        dir_okay=False,
+        readable=True,
+        allow_dash=False,
+        path_type=str,
+    ),
+    help="Path to a pyproject.toml file with specialized defaults for pip-tools",
```

Comment:
Per my comments in the issue, it is important to allow a pip-tools specific config taking preference over `pyproject.toml`. Also, the way it's implemented, it'll allow naming the file arbitrarily, so this isn't accurate with regards to the generic case, only the default. Let's try to rephrase this without implying that this filename is mandatory.

Reply:
Marking this thread as resolved.